cmd/kube-controller-manager/app/core.go

```diff
@@ -586,35 +576,44 @@ func processCIDRs(cidrsList string) ([]*net.IPNet, bool, error) {
 }
 
 // setNodeCIDRMaskSizes returns the IPv4 and IPv6 node cidr mask sizes.
-// If --node-cidr-mask-size not set, then it will return default IPv4 and IPv6 cidr mask sizes.
-func setNodeCIDRMaskSizes(cfg nodeipamconfig.NodeIPAMControllerConfiguration) (int, int, error) {
+func setNodeCIDRMaskSizes(clusterCIDRs []*net.IPNet, cfg nodeipamconfig.NodeIPAMControllerConfiguration) (int, int, error) {
```

Comment:
definitively we have to add unit tests to this function, there are too many combinations and is easy to break something in a refactor.

Actually, ??maybe in a folllow up?? we can put all the validation logic in their own function so we can add unit tests for the cluster CIDRS too?
https://github.com/kubernetes/kubernetes/blob/db71cd44966c676a9f796422968fba92ce8f44a5/cmd/kube-controller-manager/app/core.go#L105-L156

Reply:
thanks @SataQiu , the unit tests are great, personally I found them easier to understand this whole logic :)